wyvern-exchange/src/wyvern-exchange.ts

```diff
@@ -1,231 +1,367 @@
+        uints[13], //basePrice:
+        uints[14], //extra:
+        uints[15], //listingTime:
+        uints[16], //expirationTIme:
+        uints[17] //salt:
+    )
+    log.info(
+        "txHash {} sellOrder maker {} taker {} makerProtocolFee {} takerProtocolFee {} makerRelayerFee {} takerRelayerFee {} feeRecipient {} feeMethod {} side {} saleKind {} exchange {} target{}",
+        [
+            txHash,
+            sellOrder.maker,
+            sellOrder.taker,
+            sellOrder.makerProtocolFee.toString(),
+            sellOrder.takerProtocolFee.toString(),
+            sellOrder.makerRelayerFee.toString(),
+            sellOrder.takerRelayerFee.toString(),
+            sellOrder.feeRecipient,
+            sellOrder.feeMethod,
+            sellOrder.side,
+            sellOrder.saleKind,
+            sellOrder.exchange.toHexString(),
+            sellOrder.target,
+        ]
+    )
 
-    contractAddress =
-      decoded.contract != Address.zero() ? decoded.contract : contractAddress;
+    let matchPrice = orders.helpers.calculateMatchPrice(buyOrder, sellOrder, blockTimeStamp)
 
-    log.info("txHash {} transfer method {}", [txHash.toHexString(), decoded.method]);
-    let nft = new airstack.nft.NFT(contractAddress, decoded.method, decoded.token, decoded.amount);
-    let royaltyDetails = new royaltyResult();
-    if (sellOrder.feeRecipient != Address.zero().toHexString()) {
-      royaltyDetails = calculateRoyality(sellOrder.makerRelayerFee, sellOrder.feeRecipient, matchPrice);
+    if (isBundleSale) {
+        log.info("txHash {} isbundlesale", [txHash])
+        let decoded = abi.decodeBatchNftData(
+            buyOrder.callData,
+            sellOrder.callData,
+            buyOrder.replacementPattern
+        )
+        if (decoded.addressList.length != decoded.transfers.length) {
+            log.error("bundleSale addressList len != transfers len,txHash {}", [txHash])
+        }
+        let from = Address.zero()
+        let to = Address.zero()
+        let nfts: airstack.nft.NFT[] = []
+        for (let i = 0; i < decoded.transfers.length; i++) {
+            let transfer = decoded.transfers[i]
+            log.debug("from {} to {} method {} ", [
+                transfer.from.toHexString(),
+                transfer.to.toHexString(),
+                transfer.method,
+            ])
+            if (from == Address.zero()) {
+                from = transfer.from
+            } else if (from != transfer.from) {
+                log.error("from address change not expected,txhash {}", [txHash])
+                throw new Error("from address change not expected")
+            }
+            if (to == Address.zero()) {
+                to = transfer.to
+            } else if (to != transfer.to) {
+                log.error("to address change not expected,txhash {}", [txHash])
+                throw new Error("to address change not expected")
+            }
+            if (transfer.tokens.length > 1) {
```

Comment:
Need to understand this conidition

Reply:
added for verification,this never got triggered,removed the loop below